### PR TITLE
Update using-dotnet-cli.md

### PR DESCRIPTION
**Pull request by AustinWise**
The current directions for trying out your own build of coreclr are a little outdated. These edits update the directories and files names to match the locations where I was able to find the build output. I did not update the second half of the document.

### CODE REVIEW

**Comment by AustinWise** [comment]
I'd like to highlight two issues I encountered while trying this procedure.

A small one first: in a locally-built System.Private.CoreLib.dll, there is no git commit hash embedded. The nightly and preview builds do have the git hash. I don't know if this is intentional. I updated to the sample to remove the reference to the Git commit hash.

Secondly, when following this procedure, I get the following error:

```
  Determining projects to restore...
C:\test\test.csproj : error NU1102: Unable to find package Microsoft.WindowsDesktop.App.Runtime.win-x64 with version (= 5.0.0-dev)
C:\test\test.csproj : error NU1102:   - Found 25 version(s) in nuget [ Nearest version: 5.0.0-preview.1.20127.5 ]
C:\test\test.csproj : error NU1102:   - Found 0 version(s) in local CoreCLR
C:\test\test.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Runtime.win-x64 with version (= 5.0.0-dev)
C:\test\test.csproj : error NU1102:   - Found 28 version(s) in nuget [ Nearest version: 5.0.0-preview.1.20124.5 ]
C:\test\test.csproj : error NU1102:   - Found 0 version(s) in local CoreCLR
```

I was able to resolve this by editing a file in the SDK. In the Microsoft.NETCoreSdk.BundledVersions.props file, I commented out the `KnownFrameworkReference`s that mentioned in the error message. If this is not a known issue I should probably file a bug on the SDK. I did not covered this problem in my changes.

Lastly, I was going through this all to try out #40104. It looks great: according to PerfView, there are now zero foreground JIT compilations happening in hello world. The state of .NET single file packaging and AOT is much improved!

File: docs/workflow/using-dotnet-cli.md [hunk 1]
```diff
@@ -108,32 +101,48 @@ dotnet publish
 Make sure that restoring done by `dotnet publish` installed the explicit version of the Runtime that you have specified:
 
 ```
-PS C:\coreclr\helloWorld> dotnet publish
-  Restoring packages for C:\coreclr\helloWorld\helloWorld.csproj...
-  Installing runtime.win-x64.Microsoft.NETCore.Runtime.CoreCLR 3.0.0-preview1-26210-
+C:\coreclr\helloworld>dotnet publish
```
Comment (danmosemsft) [inline docs/workflow/using-dotnet-cli.md:104]: Please change `coreclr` to `runtime` as that's the repo name.

**Comment by danmosemsft** [comment]
For the missing commit SHA, I'm guessing it's intentional:
https://github.com/dotnet/arcade/blob/3c56c9f23ce0d199baef41dd6831b42a2690ebc1/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L152

File: docs/workflow/using-dotnet-cli.md (hunk 1 above)
Comment (danmosemsft) [reply docs/workflow/using-dotnet-cli.md]: Never mind I did it.

**Review by danmoseley — APPROVED** [review]

**Comment by danmosemsft** [comment]
I made some updates. The second section of this file `## Update using runtime nuget package` is still out of date and may not even be relevant anymore. Same for using-your-build.md.

**Comment by danmosemsft** [comment]
Thanks for the changes.

**Comment by danmosemsft** [comment]
https://github.com/dotnet/runtime/issues/40557

**Comment by AustinWise** [comment]
Thanks for the edits.

**Pull request merged**